### DISCUSSION TITLE
Support MJ2 (Motion JPEG 2000) file format

### DISCUSSION THREAD

**Discussion by generalciao** [talmolab/sleap#1226] [Ideas]
The .mj2 file format is produces by recent versions of Matlab, for example when streaming video data to files on disk. Currently the SLEAP GUI does not support adding videos in this format. Please add .mj2 (MJ2, Motion JPEG 2000) as a supported format. Thanks!

PS: If there is a workaround to load such videos in the current version of SLEAP, I'd appreciate some pointers on how to do so.

**Comment by roomrys** [comment]
Hi @generalciao,

We do not specifically have support for `.mj2`. I don't have a `.mj2` handy at the moment, but when adding a video, are you able to load it by selecting the any file type filter? If not, then there is not a workaround I can think of at the moment.

Thanks,
Liezl

**Comment by talmo** [comment]
Actually, you might be able to get it working if you just rename it to `.avi` -- this is another common container format for MJPEG and it's possible the OpenCV backend will just figure it out.

If not, then the workaround would be to [reencode it to an MP4 format](https://sleap.ai/help.html#does-my-data-need-to-be-in-a-particular-format).

It should be easy for us to add more native support for this though, so let's keep this as an open idea.

**Comment by generalciao** [comment]
Indeed your suggestion worked: I created a symbolic link with the .avi extension pointing to the .mj2 files and was able to add the videos to SLEAP that way (while avoiding duplicated disk space usage). Thanks!

I have a lot of files to process (and I know other people who use mj2 files as well), so native support would be preferred, in case you have time to implement it. I looked at video.py and dug around OpenCV and ffmpeg docs, but it's not clear to me what would be required in your code to avoid the "no match found for file type" exception that I encounter when directly adding .mj2 files.

**Reply by talmo** [reply]
Hi @generalciao,

Good the workaround worked!

We've created an issue (#1234) to track this, but basically it'll just require adding .mj2 to a few different places.

We might not get to adding this until the next release after 1.3.0, but if you want to take a stab at it, just try to search the code for ".mp4" or ".avi" and just add ".mj2" to those lists. Should be a quick change.

Cheers,

Talmo
